File: frontend/src/lib/models/page_elements/image_element_config.ts


import type { PageElementConfig } from "../page_element_config";

export let IMAGE_ELEMENT_TYPE = "image";

export interface ImageElementConfig extends PageElementConfig {
    url: string;
    alt?: string;
    options?: ImageElementOptions;
}

export interface ImageElementOptions {
    caption?: string;
    width?: number;
    height?: number;
}

File: src/main/kotlin/dev/schlaubi/dashboard/models/pageelements/ImageElementConfig.kt
package dev.schlaubi.dashboard.models.pageelements

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
@SerialName("image")
data class ImageElementConfig(
    val url: String, val alt: String? = null, val options: ImageElementOptions? = null
) : PageElementConfig()

@Serializable
data class ImageElementOptions(
    val width: String? = null, val height: String? = null, val caption: String? = null
)